### PR TITLE
Use a dedicated test executor in MockTransportService

### PR DESCRIPTION
**Pull request by ywangd**
Instead of using the generic executor for delayed transport actions, this PR adds a new executor to schedule these actions. It helps avoid sharing executors with the node which may lead to unexpected CI failures due to unsafe future assertion.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java [hunk 1]
```diff
@@ -302,6 +305,16 @@ private MockTransportService(
             Tracer.NOOP
         );
         this.original = transport.getDelegate();
+        this.testExecutor = EsExecutors.newScaling(
+            "mock-transport",
+            0,
+            4,
+            30,
+            TimeUnit.SECONDS,
+            false,
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java:314]: Could we set `rejectAfterShutdown` to `true` please?

File: test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java [hunk 2]
```diff
@@ -795,6 +808,8 @@ protected void doClose() throws IOException {
             }
         } catch (InterruptedException e) {
             throw new IllegalStateException(e);
+        } finally {
+            ThreadPool.terminate(testExecutor, 10, TimeUnit.SECONDS);
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java:812]: Could we have an `assertTrue` here to make sure it did actually terminate?

**Review by DaveCTurner — COMMENTED** [review]
Looks good, couple of minor nits.

File: test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java (hunk 2 above)
Comment (ywangd) [reply test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java]: Yep see 63a461498b049ba8ae0b39b924b9bc514f0484da

File: test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java (hunk 1 above)
Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java]: > it is not rejected and we should just queue it up

Sure but that's not what `false` does, it just silently drops any work submitted after shutdown.

My thinking here is twofold: firstly, we are terminating this `ExecutorService` at the end of the test, there should be no more tasks submitted by that point anyway. Secondly, IMO the silent-drop behaviour of `generic` after shutdown is a bug, even if it is unfortunately one we cannot easily address without changing lots of other things, but we can at least avoid relying on that same behaviour in specific situations like this one.

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java (hunk 1 above)
Comment (ywangd) [reply test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java]: > but that's not what false does, it just silently drops any work submitted after shutdown.

TIL. Thanks!

**Comment by ywangd** [comment]
@elasticmachine update branch

**Pull request merged**